cli/proc_state.rs

```diff
@@ -304,7 +335,7 @@ impl ProcState {
       .map(|s| (s, ModuleKind::Esm))
       .collect::<Vec<_>>();
 
-    if !reload_on_watch && !has_root_npm_specifier {
+    if !has_root_npm_specifier {
```

Comment:
I'm not sure if this branch is needed anymore, IIRC I only had problems with it because of `reload_on_watch`

Reply:
The alternative would be making this an unconditional block. I believe we want to skip this short circuiting logic if there are npm roots since `GraphData::check()` doesn't account for those (this could be cleaned up in a separate PR)